BLOOM minor changes on tokenizer

younesbelkada opened the pull request:
# What does this PR do?

- Attempts to fix minor issues with the BloomTokenizer
- Remove unused args
- Added a new attribute on `TokenizerTesterMixin` for models that are not agnostic to sequence length (typically models that use ALiBi positional embeddings)

Still need to discuss if it is worth it to force the padding side to the left

cc @SaulLu

younesbelkada committed:
few fixes:

- hardcode tokenizer padding side
- remove unused args

younesbelkada committed:
Merge branch 'main' into bloom-fix-tokenizer

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_17823). All of your documentation changes will be reflected on that endpoint.

SaulLu commented on src/transformers/models/bloom/tokenization_bloom_fast.py:
```diff
@@ -109,7 +99,7 @@ class BloomTokenizerFast(PreTrainedTokenizerFast):
     vocab_files_names = VOCAB_FILES_NAMES
     pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
-    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+    max_model_input_sizes = []
```
I am really in favor of this change, especially because with AliBi positional embeddings we don't have a max model input size (or pretrainned positional embeddings)

SaulLu replied:
Your proposition looks good to me! :smile:

younesbelkada commented on src/transformers/models/bloom/tokenization_bloom_fast.py:
```diff
@@ -123,6 +113,7 @@ def __init__(
         eos_token="</s>",
         pad_token="<pad>",
         add_prefix_space=False,
+        padding_side="left",
```
Will remove it on the next commit, I have just confirmed that we produce the same result with different padding ;)

younesbelkada replied:
Should be fixed in 261a7864dfdc2fe48b515d2ca950169030b61efa

younesbelkada committed:
few fixes:

- added new attribute on TokenizerTesterMixin
- added new slow test
- remove unused arg on tokenizer class

younesbelkada committed:
make style

SaulLu commented on src/transformers/models/bloom/tokenization_bloom_fast.py:
```diff
@@ -109,7 +99,6 @@ class BloomTokenizerFast(PreTrainedTokenizerFast):
-    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
     model_input_names = ["input_ids", "attention_mask"]
     slow_tokenizer_class = None
 
```
WDYT of adding a comment here to keep a track in the code of the reason why this tokenizer doesn't have this field fill?
```suggestion
# No `max_model_input_sizes` as BLOOM uses ALiBi positional embeddings
```

younesbelkada committed:
Update src/transformers/models/bloom/tokenization_bloom_fast.py

Co-authored-by: SaulLu <55560583+SaulLu@users.noreply.github.com>

younesbelkada committed:
make quality

younesbelkada commented:
We may merge this at the same time as https://github.com/huggingface/transformers/pull/17837 to make it a patch release of minor fixes

SaulLu reviewed (approved):
All the changes look good to me! Thanks @younesbelkada :hugs:

younesbelkada committed:
apply changes

- remove new attribute
- redefine test on the class

younesbelkada committed:
add comments

sgugger reviewed (approved):
LGTM, thanks!
